### ISSUE TITLE
Unable to use transformer-compile-class and transformer-variant-group together

### ISSUE THREAD

**Issue by so1ve** [unocss/unocss#1239]
Here is my `uno.config.ts`:

```ts
import { defineConfig, presetAttributify, presetIcons, presetUno } from "unocss";
import transformerVariantGroup from "@unocss/transformer-variant-group";
import transformerCompileClass from "@unocss/transformer-compile-class";

export default defineConfig({
   presets: [
     presetUno(),
     presetAttributify(),
     presetIcons({
       scale: 1.2,
       extraProperties: {
         "display": "inline-block",
         "height": "1.4em",
         "width": "1.4em",
         "vertical-align": "text-bottom",
       },
     }),
   ],
   transformers: [
     transformerVariantGroup(),
     transformerCompileClass(),
   ],
});
````

When running vite server, this error occurs: `[vite] Internal server error: Cannot split a chunk that has already been edited`

**Comment by antfu** [comment]
We temporarily close this due to the lack of enough information.
Please provide a [minimal reproduction](https://stackoverflow.com/help/minimal-reproducible-example) to reopen the issue.
Thanks.

> [Why reproduction is required](https://antfu.me/posts/why-reproductions-are-required)

**Comment by so1ve** [comment]
Hi antfu,
this is the reproduction.
https://stackblitz.com/edit/vitejs-vite-sn2cnh?file=uno.config.ts

**Comment by antfu** [comment]
You may just swap the two transformers
https://stackblitz.com/edit/vitejs-vite-lirvpa?file=uno.config.ts

**Comment by so1ve** [comment]
No error is occured after the swap, but no classes starting with uno- are generated.

**Comment by so1ve** [comment]
A syntax error occurs. It seems that this will destroy the original jsx file. like:

![](https://user-images.githubusercontent.com/58381667/178150829-3705bfaa-e6db-4527-b825-f64437cd11f3.png)

**Comment by so1ve** [comment]
Sometimes the error does not occur, but the variant group cannot be converted.
![](https://user-images.githubusercontent.com/58381667/178150883-3e4890c6-720b-4720-a8e4-5b09bfcd79d3.png)

**Comment by antfu** [comment]
Should be fixed in 0.44.1

**Comment by mklueh** [comment]
@antfu I'm running into the same issue

> "@unocss/nuxt": "^0.46.3",


```
 ERROR  [unocss:transformers:default] Cannot split a chunk that has already been edited (0:666 – "@screen sm{.markdown blockquote,.markdown h2,.markdown h3,.markdown h4,.markdown h5,.markdown h6{@apply ")
file: /home/user/workspace/my-project/apps/my-app/assets/scss/_markdown.scss?vue&type=style&index=0&src=true&lang.scss
 ERROR  Cannot split a chunk that has already been edited (0:666 – "@screen sm{.markdown blockquote,.markdown h2,.markdown h3,.markdown h4,.markdown h5,.markdown h6{@apply ")
  at MagicString._splitChunk (/home/user/workspace/my-project/node_modules/magic-string/dist/magic-string.es.mjs:919:10)
  at MagicString._split (/home/user/workspace/my-project/node_modules/magic-string/dist/magic-string.es.mjs:909:43)
  at MagicString.remove (/home/user/workspace/my-project/node_modules/magic-string/dist/magic-string.es.mjs:788:8)
  at parseApply (/home/user/workspace/my-project/node_modules/@unocss/transformer-directives/dist/index.cjs:164:8)
  at async List.<anonymous> (/home/user/workspace/my-project/node_modules/@unocss/transformer-directives/dist/index.cjs:106:7)
  at async Promise.all (index 0)
  at async handleApply (/home/user/workspace/my-project/node_modules/@unocss/transformer-directives/dist/index.cjs:102:3)
  at async processNode (/home/user/workspace/my-project/node_modules/@unocss/transformer-directives/dist/index.cjs:202:7)
  at async Promise.all (index 184)
  at async transformDirectives (/home/user/workspace/my-project/node_modules/@unocss/transformer-directives/dist/index.cjs:205:3)
  at async applyTransformers (/home/user/workspace/my-project/node_modules/@unocss/vite/dist/index.mjs:978:5)
  at async transform (/home/user/workspace/my-project/node_modules/rollup/dist/es/shared/rollup.js:21734:16)
  at async ModuleLoader.addModuleSource (/home/user/workspace/my-project/node_modules/rollup/dist/es/shared/rollup.js:21959:30)

 ——————————————————————————————————————————
```

with this stylesheet (that worked previously with TailwindCSS)

```

.markdown {

  #table-of-contents + ul {
    @apply shadow py-2;

    li {
      @apply m-4 list-none;
    }
  }

  p {
    img {
      @apply w-full h-full object-fill;
    }
  }

  p, ul, ol {
    @apply mb-4;
  }

  ul li, ol li {
    @apply mb-4;
  }


  ul {
    @apply list-disc pl-4;
  }

  ol {
    @apply list-decimal pl-4;
  }

  h2, h3, h4, h5, h6 {
    @apply font-sans font-bold text-gray-800  mb-6 mt-8;
  }

  h5, h6 {
    @apply uppercase;
  }

  h2 {
    @apply text-xl;
  }

  h3 {
    @apply text-lg;
  }

  h4 {
    @apply text-lg;
  }

  h5 {
    @apply text-base;
  }

  h6 {
    @apply text-sm;
  }

  @screen sm {
    h2 {
      @apply text-4xl;
    }
  }

  a {

    &:not(h1, h2, h3, h4, h5, h6) {
      color: #00aabe;
      @apply break-words;

      & :hover {
        @apply text-blue-700;
      }
    }
  }

  pre {
    @apply rounded  #{!important};
    @apply shadow-inner -mx-2 p-6 my-4 text-sm overflow-x-auto;
  }

  pre > code{
    @apply bg-yellow-200
  }

  mark {
    @apply bg-yellow-300 px-2;
  }

  dl {

    dt {
      @apply text-black text-sm font-sans font-bold uppercase mb-2;

      + dd {
        @apply mb-4 pl-2;
      }

    }

    dd {
      @apply mb-4;
    }

  }

  blockquote {
    @apply p-2 py-6 mb-4 mx-2 italic;

    p {
      @apply mb-0 pl-2 border-l-4 border-blue-700;
    }

  }

  hr {
    @apply h-px bg-gray-300 my-8;
  }

}

@screen sm {
  .markdown {

    blockquote {
      @apply mx-2;
    }

    h2 {
      @apply text-4xl;
    }

    h3 {
      @apply text-3xl;
    }

    h4 {
      @apply text-2xl;
    }

    h5 {
      @apply text-xl;
    }

    h6 {
      @apply text-base;
    }
  }
}

.post-cover {

  & ::after {
    @apply absolute w-full h-48 left-0 bottom-0 z-0;
    content: '';
    background: linear-gradient(to bottom, transparent 0%, rgba(0, 0, 0, .5) 100%);
  }

  & .fullscreen::after {
    @apply h-full;
    background: rgba(0, 0, 0, .15);
  }

}

```


```
import {
  type UserConfig,
  defineConfig,
  presetAttributify,
  presetIcons,
  presetTypography,
  presetUno,
  presetWebFonts,
  transformerCompileClass,
  transformerDirectives,
  transformerVariantGroup,
} from 'unocss'


export default defineConfig({
  shortcuts: [
    [
      'input-base',
      'appearance-none rounded-none border border-gray-500 bg-transparent px-3 py-2 text-base shadow-none sm:text-sm focus:(border-blue-600 outline outline-2 outline-offset-2 outline-transparent ring-1 ring-blue-600 ring-offset-0 ring-offset-transparent)',
    ],
    [
      'button-base',
      'inline-flex items-center justify-center rounded-md border border-gray-300 px-4 py-2 text-sm font-medium text-gray-700 shadow-sm hover:bg-gray-50 focus:(outline-none ring-2 ring-blue-500)',
    ],
    [
      'button-primary',
      'border-transparent bg-blue-600 text-white hover:bg-blue-700 focus:ring-blue-500',
    ],
    [
      'button-secondary',
      'border-transparent bg-red-600 text-white hover:bg-red-700 focus:ring-red-500',
    ],


    ['card-base', 'bg-white overflow-hidden shadow rounded-lg'],
    ['card-body', 'px-4 py-5 sm:p-6'],
    ['card-footer', 'bg-gray-50 px-4 py-4 sm:px-6'],
  ],

  presets: [
    presetUno(),
    presetAttributify(),
    presetIcons({
      prefix: '',
      scale: 1.2,
      extraProperties: {
        display: 'inline-block',
        'vertical-align': 'middle',
      },
    }),
    presetTypography(),
    presetWebFonts({
      fonts: {
        sans: 'DM Sans',
        serif: 'DM Serif Display',
        mono: 'DM Mono',
      },
    }),
  ],
  transformers: [
    transformerCompileClass(),
    transformerVariantGroup(),
    transformerDirectives()
  ],
} as UserConfig)

```

I've swapped the transformers back and forth without any effect
